server/channels/api4/post.go

```diff
@@ -652,7 +665,17 @@ func deletePost(c *Context, w http.ResponseWriter, _ *http.Request) {
 		}
 	}
 
-	if _, err := c.App.DeletePost(c.AppContext, c.Params.PostId, c.AppContext.Session().UserId); err != nil {
+	if permanent {
+		if *c.App.Config().ServiceSettings.EnableAPIPostDeletion {
+			err = c.App.PermanentDeletePost(c.AppContext, c.Params.PostId, c.AppContext.Session().UserId)
+		} else {
+			err = model.NewAppError("deletePost", "api.post.delete_post.not_enabled.app_error", nil, "postId="+post.Id, http.StatusUnauthorized)
```

Comment:
2/5 that this should be a 501. We use the error code in other parts of the code when a config setting is missing, e.g. in https://github.com/mattermost/mattermost/blob/aa85a13c8f0d6a5cfd24f90e7b529076055d2638/server/channels/api4/drafts.go#L73-L77.